src/components/evaluation/EvaluationTable.vue

```diff
@@ -532,13 +536,29 @@ export default {
     getEvaluationTypes().then(data => {
       this.evaluationTypes = [{id: null, name: 'None'}].concat(data)
     })
-    this.rules.currentTermDate = v => {
-      const formatted = this.$moment(v).format('YYYY-MM-DD')
-      if (formatted > this.$config.currentTermDates.begin && formatted < this.$config.currentTermDates.end) {
-        return true
+    this.rules.currentTermDate = (selectedDate, sisStartDate, sisEndDate) => {
+      const formattedSelectedDate = this.$moment(selectedDate).format('YYYY-MM-DD')
+      const formattedCourseStart = this.$moment(sisStartDate).format('YYYY-MM-DD')
+      const formattedCourseEnd = this.$moment(sisEndDate).format('YYYY-MM-DD')
+
+      const courseLength = this.$moment(formattedCourseEnd).diff(formattedCourseStart, 'days')
+      let lastStartDate =  this.$moment(sisEndDate)
+
+      if (this.selectedTerm.name.includes('Summer')) {
+          lastStartDate = courseLength < 90 ? lastStartDate.subtract(13, 'day') : lastStartDate.subtract(20, 'day')
+          return formattedSelectedDate <= lastStartDate.format('YYYY-MM-DD') ? true : 'Date must be within current term.'
+      } else {
+        
+        if (formattedCourseEnd <= this.$config.currentTermDates.end) {
```

Comment:
@cesarvh This logic all looks correct to me, except that the value in `this.$config.currentTermDates.end` is just a loose value that we implemented for first-pass sanity checking and not the default meeting end date for the term. (See damien.lib.berkeley.current_term_dates, which is the server-side source for these values.) What we actually want to test against are the default meeting dates for the term as provided by damien.lib.queries.get_default_meeting_dates; you might need to add these values to the config feed.